hvac/adapters.py

```diff
@@ -366,6 +366,8 @@ def request(self, *args, **kwargs):
                 return response.json()
             except ValueError:
                 pass
+        elif response.status_code == 204:
+            return {}
 
         return response
```

Comment:
We should probably never return the response object. Do we also return an empty dictionary here? Raise an exception?

Reply:
Hm, I tend to agree that returning the response is not the best idea, though it has been in the adapter since it was introduced: https://github.com/hvac/hvac/pull/537/files#diff-db919c05d0fa65b24b7b140bab3fd3530b3530735e4e9e20a843881557158936

There is a `RawAdapter` that can be used when the full response object is desired, and that's the superclass for this already. 

I suppose the way we can look at this is that this is, specifically, the `JSONAdapter`, so a response that's not valid or parseable as JSON could (should?) be an exception. 

I don't think it should be an empty dictionary in this case, because it implies that everything went ok but really there was a response there that's now hidden.

The main thing we need to be careful of is that this is the default adapter, but then again, if anyone was using this full response object I'd guess it was mostly to deal with 204s, since 4xx and 5xx codes would already be raising exceptions.